manim/utils/bezier.py

```diff
@@ -1012,11 +1019,13 @@ def interpolate(start: float, end: float, alpha: ColVector) -> ColVector: ...
 
 
 @overload
-def interpolate(start: Point3D, end: Point3D, alpha: float) -> Point3D: ...
+def interpolate(start: Point3DLike, end: Point3DLike, alpha: float) -> Point3D: ...
```

Comment:
I actually don't think `Point3DLike` is right here, because it performs `start + end * (1-alpha)`, which isn't possible for e.g. a `tuple`.

I think a similar logic might apply for other changes in this file?

Reply:
You're right, I missed that. There are many other parameters in this file which should also be typed as `Point3D`.